core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for Oppia feedback reports from both Android and web."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+
+from core.platform import models
+import feconf
+import python_utils
+import utils
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+datastore_services = models.Registry.import_datastore_services()
+transaction_services = models.Registry.import_transaction_services()
+
+PLATFORM_CHOICE_ANDROID = 'android'
+PLATFORM_CHOICE_WEB = 'web'
+PLATFORM_CHOICES = [PLATFORM_CHOICE_ANDROID, PLATFORM_CHOICE_WEB]
+
+REPORT_INFO_TO_REDACT = [
```

Comment:
Use a tuple. Ditto above.

Tuples are unmodifiable, hence preferred. Can also use frozenset if you need fast lookups (probably unnecessary in this case).

Reply:
Done